modules/OCPPConfiguration/main/event_handler.hpp

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+
+#pragma once
+
+#include "generated/types/ocpp.hpp"
+#include "mapping_reader.hpp"
+
+namespace module {
+
+class EventHandler {
+public:
+    EventHandler(const std::filesystem::path& config_mapping_path);
+    void handle_event(const types::ocpp::EventData& event_data, const std::string& user_config_path_string);
+
+private:
+    const std::optional<EverestConfigMapping>
+    find_mapping_by_component_variable_or_log_error(const types::ocpp::ComponentVariable& component_variable) const;
```

Comment:
It should be considered normal to receive event data for variables that are not part of the mapping, so I wouldnt throw a `runtime_error` exception here but simply do nothing in case nothing can be found in the mapping. I think its enough to only have this function:

`const std::optional<EverestConfigMapping>
    find_mapping_by_component_variable(const types::ocpp::ComponentVariable& component_variable) const;`

Reply:
I decided to keep both functions but change the high level function to not log any error.    
My reasoning was that the high level function checks if the low level function returned a value and if not returns a nullopt and the other function makes the actual lookup.    
This way both functions have just one responsibility and I can still use the raii approach in the caller function of the high level function